src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs

```diff
@@ -690,8 +690,8 @@ internal async Task<List<string>> GetOrderedAliasesAsync(string authority, bool
                         IdToken.TryParse(result.Result.IdToken, out IdToken idToken);
 
                         CacheFallbackOperations.WriteMsalRefreshToken(TokenCacheAccessor, result, authority, clientId, displayableId,
-                            result.Result.UserInfo.GivenName,
-                            result.Result.UserInfo.FamilyName,
+                            result.Result.UserInfo?.GivenName,
+                            result.Result.UserInfo?.FamilyName,

```

Comment:
adding ? is the actual fix. the customer was getting a null ref because UserInfo is null here. There will no longer be a null ref here. 
![image](https://user-images.githubusercontent.com/19942418/58603799-f61e1900-8246-11e9-8f24-b4891a9ce954.png)

Reply:
Will caching still work? Is it then possible to complete the scenario?

Reply:
you are right. if there is no id_token, then we will exit out of writing items to the MSAL cache...[see here]([https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/blob/dev/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Core/Cache/CacheFallbackOperations.cs#L52)...but](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/blob/dev/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Core/Cache/CacheFallbackOperations.cs#L52)...but) i'm assuming, since this has been out for awhile, that it hasn't been very impactful, or our customers are not using the cache in that way?

---
In reply to: [288842460](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1607#discussion_r288842460) [](ancestors = 288842460)

Reply:
addressed the comments. lmk if you'd like to discuss anything further. thanks for the utest suggestions and improvements.

---
In reply to: [288855740](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1607#discussion_r288855740) [](ancestors = 288855740,288842460)